lib/utils/docker.coffee

```diff
@@ -131,210 +123,56 @@ exports.generateConnectOpts = generateConnectOpts = (opts) ->
-							'''
-							return tarStream
-						else
-							[arch, deviceType] = info
-							# Perform type resolution on the project
-							bundle = new resolver.Bundle(tarStream, deviceType, arch)
-							resolver.resolveBundle(bundle, resolver.getDefaultResolvers())
-							.then (resolved) ->
-								logger.logInfo("Building #{resolved.projectType} project")
-
-								return resolved.tarStream
-					.then (buildStream) ->
-						# if we need emulation
-						if options.emulated and platformNeedsQemu()
-							return transpose.transposeTarStream buildStream,
-								hostQemuPath: toPosixPath(qemuPath)
-								containerQemuPath: "/tmp/#{QEMU_BIN_NAME}"
-						else
-							return buildStream
-					.then (buildStream) ->
-						# Send the resolved tar stream to the docker daemon
-						buildStream.pipe(stream)
-					.catch(reject)
-
-					# And print the output
-					logThroughStream = es.through (data) ->
-						logs += data.toString()
-						this.emit('data', data)
-
-					if options.emulated and platformNeedsQemu()
-						buildThroughStream = transpose.getBuildThroughStream
-							hostQemuPath: toPosixPath(qemuPath)
-							containerQemuPath: "/tmp/#{QEMU_BIN_NAME}"
-
-						newStream = stream.pipe(buildThroughStream)
-					else
-						newStream = stream
-
-					newStream
-					.pipe(logThroughStream)
-					.pipe(cacheHighlightStream())
-					.pipe(logger.streams.build)
-
-			# Create a builder
-			generateConnectOpts(options)
-			.tap (connectOpts) ->
-				ensureDockerSeemsAccessible(connectOpts)
-			.then (connectOpts) ->
-				# Allow degugging output, hidden behind an env var
-				logger.logDebug('Connecting with the following options:')
-				logger.logDebug(JSON.stringify(connectOpts, null, '  '))
-
-				builder = new dockerBuild.Builder(connectOpts)
-				opts = {}
-
-				if options.tag?
-					opts['t'] = options.tag
-				if options.nocache?
-					opts['nocache'] = true
-				if options.buildArg?
-					opts['buildargs'] = parseBuildArgs options.buildArg, (arg) ->
-						logger.logWarn("Could not parse variable: '#{arg}'")
-				if options.squash?
-					opts['squash'] = true
-
-				builder.createBuildStream(opts, hooks, reject)
-
-# Given an image id or tag, export the image to a tar archive,
-# gzip the result, and buffer it to disk.
-exports.bufferImage = (docker, imageId, bufferFile) ->
-	Promise = require('bluebird')
-	streamUtils = require('./streams')
-
-	image = docker.getImage(imageId)
-	imageMetadata = image.inspectAsync()
-
-	Promise.join image.get(), imageMetadata.get('Size'), (imageStream, imageSize) ->
-		streamUtils.buffer(imageStream, bufferFile)
-		.tap (bufferedStream) ->
-			bufferedStream.length = imageSize
+exports.generateBuildOpts = (options) ->
+	opts = {}
+	if options.tag?
+		opts['t'] = options.tag
+	if options.nocache?
+		opts['nocache'] = true
+	if options.squash?
+		opts['squash'] = true
+	if options.buildArg?
+		opts['buildargs'] = parseBuildArgs(options.buildArg)
```

Comment:
Why are you using the `['name']` form rather than `obj.name = ` ?

Reply:
It was like this before and didn't think changing it as it felt natural to me given the object is treated as a dictionary here. Have no strong opinion though, either way is fine by me, will change.